src/basic/strv.c

```diff
@@ -66,6 +66,36 @@ char* strv_find_startswith(char * const *l, const char *name) {
         return NULL;
 }
 
+char* strv_find_closest(char * const *l, const char *name) {
+        ssize_t best_distance = SSIZE_MAX;
+        char *best = NULL;
+
+        assert(l);
```

Comment:
we generally treat a NULL strv like an empty strv. Hence please do not add an assert like this